src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+import { useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import './Autocomplete.scss';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [focusOnInput, setFocusOnInput] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [query, setQuery] = useState('');
+
+  const applyQuery = debounce(setAppliedQuery, delay);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+    applyQuery(value);
+    setFocusOnInput(true);
+    onSelected(null);
+  };
+
+  const handleSelectPerson = (person: Person) => {
```

Comment:
The naming convention for this method could be improved. According to the [conventions](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364), it's better to name the function `onSelectPerson` instead of `handleSelectPerson`.